File: frontend/lib/common/models/ingredient_preview.dart
class IngredientPreview {
  IngredientPreview(
      {required this.id,
      required this.name,
      required this.image,});
  final int id;
  final String name;
  final String image;

  factory IngredientPreview.fromJson(Map<String, dynamic> json) {
    return IngredientPreview(
        id: json['id'],
        name: json['name'],
        image: json['image']
    );
  }
}

File: backend/src/main/java/com/example/backend/entity/IngredientPreview.java
package com.example.backend.entity;

import lombok.Getter;
import org.json.JSONObject;

import static com.example.backend.config.Constants.SPOONACULAR_INGREDIENT_IMAGE_URL;

@Getter
public class IngredientPreview {
    private final Integer id;
    private final String name;
    private final String image;

    public IngredientPreview(JSONObject json) {
        id = json.getInt("id");
        name = json.getString("name");

        image = SPOONACULAR_INGREDIENT_IMAGE_URL + json.getString("image");
    }
}
